bundle/src/main/java/com/adobe/acs/commons/util/ResourceUtil.java

```diff
@@ -132,30 +120,14 @@ public static String getPropertyString(Resource resource, String namePattern) th
      * @param namePattern Property name.
      * @return Property values.
      */
-    public static List<String> getPropertyStrings(Resource resource, String namePattern) throws RepositoryException {
-        List<String> props = new ArrayList<>();
-        for (Value val : getPropertyValues(resource, namePattern)) {
-            props.add(val.getString());
-        }
-        return props;
-    }
-
-    /**
-     * Internal method to get all values for a multi-value property.
-     *
-     * @param resource The resource from which to get the property.
-     * @param namePattern Property name.
-     * @return Property values.
-     */
-    private static Value[] getPropertyValues(Resource resource, String namePattern) throws RepositoryException {
-        Property prop = getProperty(resource, namePattern);
-        if (prop != null) {
-            if (prop.isMultiple()) {
-                return prop.getValues();
-            } else {
-                return (new Value[]{prop.getValue()});
+    public static List<String> getPropertyStrings(Resource resource, String namePattern) {
+        String[] vals = resource.getValueMap().get(namePattern, String[].class);
+        List<String> valsList = new ArrayList<>();
```

Comment:
what about just using Arrays.asList() here? No reason (at least none that I can think of) for the resulting list to be mutable.

Reply:
ya...I always shy away from making lists immutable and blindsiding someone with an error if they have a reason to try to add to the list, since the `List` interface really gives no indication of whether a list can be modified or not.  However, since all other data types returned from properties are immutable, I can see going that way as well.